File: Java/String/Reverse a String.java
//https://www.geeksforgeeks.org/problems/reverse-a-string/1
class Solution {
    public static String reverseString(String str) {
        // code here
        char[] chars = str.toCharArray();
        int i = 0, j = chars.length - 1;

        // Reverse the characters in the array
        while (i < j) {
            char temp = chars[i];
            chars[i] = chars[j];
            chars[j] = temp;
            i++;
            j--;
        }

        // Convert the character array back to string
        return new String(chars);
    }
}

File: Python/String/Reverse a String.py
//https://www.geeksforgeeks.org/problems/reverse-a-string/1
class Solution:
    def reverseString(self, s: str) -> str:
        # Convert the string to a list to modify it
        s = list(s)
        i, j = 0, len(s) - 1
        
        while i <= j:
            # Swap the characters
            s[i], s[j] = s[j], s[i]
            i += 1
            j -= 1
            
        # Convert the list back to a string
        return ''.join(s)
